### PR TITLE
LPS-165994 | master

### PR DESCRIPTION
**Pull request by kristinonias**
https://issues.liferay.com/browse/LPS-165994

### CODE REVIEW

**Comment by epark18** [comment]
@kristinonias hey could you rebase this when you have a chance? Thank you!

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 1]
```diff
@@ -1158,6 +1158,68 @@ definition {
...
+				fieldLabel = "Custom Field",
+				fieldType = "Date");
+
+			Refresh();
+
+			ObjectAdmin.assertObjectField(
+				fieldLabel = "Custom Field",
+				fieldType = "Date");
+
+			Refresh();
```
Comment (epark18) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1188]: This Refresh should not be needed

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 2]
```diff
@@ -1158,6 +1158,68 @@ definition {
...
+
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object 165994");
```
Comment (epark18) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1192]: If you are publishing the object via API, you can remove these steps that navigate to the objects portlet.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 3]
```diff
@@ -1158,6 +1158,68 @@ definition {
...
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject165994");
+		}
+
+		task ("When an entry is created") {
```
Comment (epark18) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1197]: Edit this task description so it mentions that the entry should be invalid

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 4]
```diff
@@ -1158,6 +1158,68 @@ definition {
...
+
+			LexiconEntry.gotoAdd();
+
+			WaitForPageLoad();
+
+			Type(
+				locator1 = "ObjectPortlet#VIEW_EMPTY_LABEL_IN_DATE_ENTRY",
+				value1 = "23/01/2020");
+
+			WaitForPageLoad();
```
Comment (epark18) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1211]: Is this second Wait needed?

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 5]
```diff
@@ -1158,6 +1158,68 @@ definition {
...
+
+			Button.clickSave();
```
Comment (epark18) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1213]: PortletEntry.save(); is better here

**Comment by andreefarias** [comment]
Hey @kristinonias,

Could resolve the conflicts?

**Comment by kristinonias** [comment]
> 

Sorry about the delay @andreefarias I was on PTO. I've made the corrections and there should be no conflicts

**Comment by andreefarias** [comment]
Good work @kristinonias, it looks good to me! 🎉 
Thanks for your help!

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 6]
```diff
@@ -1268,6 +1268,58 @@ definition {
...
+				fieldType = "Date");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject165994");
+		}
+
+		task ("When an entry is created and the date is invalid") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Object",
+				panel = "Control Panel",
+				portlet = "Custom Objects 165994");
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1303]: Could use the macro: `ObjectAdmin.goToCustomObject(objectName = "CustomObject165994");`

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 7]
```diff
@@ -1268,6 +1268,58 @@ definition {
 		}
 	}
 
+	@description = "LPS-165994 - Verify that the user cannot insert an invalid date on Start / End fields when creating a filter"
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1271]: You could use the LPS from the story ticket here: `LPS-156704`

Comment (andreefarias) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1271]: Change the `LPS-165994` (ticket number) to `LPS-156704`

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 8]
```diff
@@ -1268,6 +1268,58 @@ definition {
...
+	@priority = "5"
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1272]: On the test map and QA validation, this testcase was defined as `@priority = "2"`

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 9]
```diff
@@ -1268,6 +1268,58 @@ definition {
...
+	test CannotInsertInvalidDate {
+		property portal.acceptance = "true";
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1274]: Delete it. Because will be a low priority.

**Review by andreefarias — CHANGES_REQUESTED** [review]
Hi @kristinonias,

Thanks a lot for your contribution. There are minor fixes that I would like to see fixed. Could you see it when you have some time?

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase (hunk 6 above)
[the remark this answers is not in the document]
Comment (kristinonias) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase]: Changed to ObjectAdmin.goToCustomObject macro.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase (hunk 7 above)
Comment (kristinonias) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase]: Changed the LPS in the description.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase (hunk 8 above)
Comment (kristinonias) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase]: Changed priority to 2.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase (hunk 9 above)
Comment (kristinonias) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase]: Not sure what I needed to remove here, but the comment above, I changed the priority.

**Comment by kristinonias** [comment]
Hi @andreefarias 

I made the changes, as well as I noticed the test did not have the filter portion included in this PR.  It's been added recently by me if you could review that as well. Thank you!

**Comment by epark18** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 10]
```diff
@@ -1212,6 +1212,75 @@ definition {
 		}
 	}
 
+	@description = "This is a use case for LPS-156704 - Verify that the user cannot insert an invalid date on Start / End fields when creating a filter"
+	@priority = "2"
+	test CannotInsertInvalidDate {
+		property portal.acceptance = "true";
```
Comment (epark18) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1218]: This does not need to be on acceptance.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 11]
```diff
@@ -1212,6 +1212,75 @@ definition {
...
+				fieldType = "Date");
+
+			ObjectAdmin.assertObjectField(
+				fieldLabel = "Custom Field",
+				fieldType = "Date");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject165994");
+		}
+
+		task ("And a new Filter is added") {
```
Comment (epark18) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1243]: Not needed after discussion

Comment (kristinonias) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase]: Removed this portion.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase (hunk 10 above)
Comment (kristinonias) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase]: Removed this portion.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase (hunk 9 above)
Comment (kristinonias) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase]: No worries, removed acceptance and changed priority back to 5.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 12]
```diff
@@ -1214,6 +1214,53 @@ definition {
 		}
 	}
 
+	@description = "This is a use case for LPS-156704 - Verify that the user cannot insert an invalid date in an object field."
+	@priority = "5"
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1218]: Here stays with `@priority = "2"`. Could you change it again?

```
@description = "LPS-156704 - Verify that the user cannot insert an invalid date in an object field."
@priority = "2"
test CannotInsertInvalidDate {

```

I'll describe it better in the next review. Thanks for your patience. 😄 
cc: @kristinonias

Comment (kristinonias) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase]: No worries! Changed back to priority 2.

**Review by andreefarias — APPROVED** [review]

**Comment by epark18** [comment]
ci:forward

**Pull request closed without merging**